sdk/core/Azure.Core/src/Shared/Base64Url.cs

```diff
@@ -36,23 +34,22 @@ internal static string DecodeString(string encoded)
             return UTF8Encoding.UTF8.GetString(Decode(encoded));
         }
 
-        /// <summary>Converts a Base64URL encoded string to a byte array.</summary>
-        /// <param name="encoded">The Base64Url encoded string.</param>
-        /// <returns>The byte array represented by the Base64URL encoded string.</returns>
-        internal static byte[] Decode(string encoded)
+        /// <summary>Encode a string as a Base64URL encoded string.</summary>
+        /// <param name="value">String input buffer.</param>
+        /// <returns>The UTF8 bytes for the string, encoded as a Base64URL string.</returns>
+        internal static string EncodeString(string value)
         {
-            encoded = encoded.Replace('-', '+').Replace('_', '/');
-            encoded = FixPadding(encoded);
-            return Convert.FromBase64String(encoded);
+            return Encode(UTF8Encoding.UTF8.GetBytes(value));
         }
 
-        /// <summary>Adds missing padding to a Base64 encoded string.</summary>
-        /// <param name="unpadded">The unpadded input string.</param>
-        /// <returns>The padded string.</returns>
-        private static string FixPadding(string unpadded)
```

Comment:
Why isn't this method required anymore?

Reply:
FixPadding is folded into the new Base64Url class. It was only ever needed because of the implementation of Base64Url::Encode (which needs to add padding). Since nothing else uses it, it can be deleted IMHO.

Note: Scott owns the pull request, I just wrote this code :)

Reply:
I took the Key Vault implementation to start, the diff engine just marked this as a rename of the Attestation service's implementation since I used it's XML comments.  As Larry mentioned in the ported implementation this is handled inline by `Append('=', (encoded.Length % 4 == 0) ? 0 : 4 - (encoded.Length % 4))`

```c#
encoded = new StringBuilder(encoded).Replace('-', '+').Replace('_', '/').Append('=', (encoded.Length % 4 == 0) ? 0 : 4 - (encoded.Length % 4)).ToString();
```